realis.go

```diff
@@ -522,6 +522,50 @@ func (c *Client) RestartInstances(key aurora.JobKey, instances ...int32) error {
 	return nil
 }
 
+// Restarts specific instances specified with slaAware.
+// If slaPolicy is nil, uses the existing slaPolicy of taskConfig.
+func (c *Client) SlaRestartInstances(slaPolicy *aurora.SlaPolicy, jobKey *aurora.JobKey, instances ...int32) error {
+	c.logger.DebugPrintf("SlaRestartInstances Thrift Payload: %v %+v %v\n", slaPolicy, jobKey, instances)
+
+	if len(instances) == 0 {
+		c.logger.DebugPrintf("it is not necessary to restart 0 instances")
+		return nil
+	}
+
+	jobSummary, err := c.GetJobSummary(jobKey.Role)
```

Comment:
Looking at the code for GetJobSummary on the scheduler side, this should be ok but we could improve on the amount of data being transferred. 

I'm gonna make a PR to add support for the `getConfigSummary` API.
From the source code that should allow us to fetch a specific instance's config:
https://github.com/aurora-scheduler/scheduler/blob/ccf7eddbfe7bf8fc0ece01baec0c048cf44fa1df/src/main/java/org/apache/aurora/scheduler/thrift/ReadOnlySchedulerImpl.java#L215 

Then we can use that here in lieu of GetJobSummary.

Any objections to that @lenhattan86 ?

Reply:
Discussion offline:

getConfigSummary doesn't have all the info we need to create the update call.

@lenhattan86  Add a test and we're good to go.